lib/dep-graph-builders/pnpm/lockfile-parser/lockfile-parser.ts

```diff
@@ -96,9 +111,49 @@ export abstract class PnpmLockfileParser {
       ? this.normalizeTopLevelDeps(root.peerDependencies || {}, false, importer)
       : {};
 
+    if (importer) {
+      this.extractedPackages[`${pkgName}@${pkgVersion}`] = {
+        id: `${pkgName}@${pkgVersion}`,
+        name: pkgName,
+        version: pkgVersion,
+        dependencies: this.topLevelDepsToNormalizedPkgs(prodDeps),
+        devDependencies: this.topLevelDepsToNormalizedPkgs(devDeps),
+        optionalDependencies: this.topLevelDepsToNormalizedPkgs(optionalDeps),
+        isDev: false,
+      };
+    }
     return { ...prodDeps, ...devDeps, ...optionalDeps, ...peerDeps };
   }
 
+  public normalizedPkgToTopLevel(pkg: NormalisedPnpmPkg): PnpmDeps {
```

Comment:
nitpick-ish: names that begin past tense such as `normalizedBlaBla` are usually used for variables, constants, things that don't compute. For function is common to use present verbs like `normalizeBlaBla`

Reply:
this function is basically just an object transformation from a format (the format for normalised packages) another format (PnpmDeps)